backend/src/main/java/com/bakdata/conquery/models/events/stores/specific/DateRangeTypeCompound.java

```diff
@@ -0,0 +1,156 @@
+public class DateRangeTypeCompound implements DateRangeStore {
+
+
+	@NotNull
+	@NotEmpty
+	private String startColumn, endColumn;
+
+	/**
+	 * does not have to be serialized because will be lazy-loaded after the deserialization of DateRangeTypeCompound
+	 *
+	 * @implNote since this value is lazy loaded, do not use it directly and use the getter instead.
+	 */
+	@JsonIgnore
+	private DateStore startStore;
+
+	/**
+	 * does not have to be serialized because will be lazy-loaded after the deserialization of DateRangeTypeCompound
+	 *
+	 * @implNote since this value is lazy loaded, do not use it directly and use the getter instead.
+	 */
+	@JsonIgnore
+	private DateStore endStore;
+
+
+	/**
+	 * per line store if line is present.
+	 */
+	private BooleanStore has;
+
+
+	/**
+	 * Represents the bucket from where this will be saved
+	 * This bucket will be set after the deserialization
+	 */
+	@Setter(AccessLevel.PROTECTED)
+	@JsonIgnore
+	private Bucket parent;
+
+	public DateRangeTypeCompound(String startColumn, String endColumn, BooleanStore has) {
+		setStartColumn(startColumn);
+		setEndColumn(endColumn);
+		setHas(has);
+	}
+
+	@JsonIgnore
+	public DateStore getStartStore() {
+		if (startStore == null) {
+			setStartStore((DateStore) getParent().getStore(getStartColumn()));
+		}
+		return startStore;
+	}
+
+	@JsonIgnore
+	public DateStore getEndStore() {
+
+		if (endStore == null) {
+			setEndStore((DateStore) getParent().getStore(getEndColumn()));
+		}
+		return endStore;
+	}
+
+	@Override
+	public void setParent(@NonNull Bucket bucket) {
+		parent = bucket;
+	}
+
+	@Override
+	public int getLines() {
+		return has.getLines();
+
+	}
+
+	/**
+	 * Estimated number of bits required to store a value of type {@link DateRangeTypeCompound}.
+	 *
+	 * @return always 0 because this store does not hold data of its own, but references its neighbouring stores.
+	 */
+	@Override
+	public long estimateEventBits() {
+		return has.estimateEventBits();
+	}
+
+	@Override
+	public DateRangeTypeCompound select(int[] starts, int[] length) {
+		return new DateRangeTypeCompound(getStartColumn(), getEndColumn(), has.select(starts, length));
+	}
+
+	@Override
+	public void setDateRange(int event, CDateRange raw) {
+		// this is already done by the child stores, so no need to do it again
+	}
+
+	@Override
+	public void setNull(int event) {
+		has.setNull(event);
+	}
+
+	public void setHas(int event, boolean present) {
+		has.setBoolean(event, present);
+	}
+
+	@Override
+	public CDateRange getDateRange(int event) {
+		int start = Integer.MIN_VALUE;
+		int end = Integer.MAX_VALUE;
+
+		final DateStore startStore = getStartStore();
+		final DateStore endStore = getEndStore();
+
+		if (startStore.has(event)) {
+			start = startStore.getDate(event);
+		}
+
+		if (endStore.has(event)) {
+			end = endStore.getDate(event);
+		}
+
+		return CDateRange.of(start, end);
+	}
+
+	@Override
+	public boolean has(int event) {
+		return has.getBoolean(event);
```

Comment:
warum nicht `has.has(event)` ?

Reply:
die has-Variable speichert die Events die gültig bzw. in dem Store zu finden sind. Der Variable-Name `has` verwirrt ja ein bisschen an der Stelle. :-)

Reply:
ich denke das ist das gleiche. Die getBoolean holt den Wert von values und bei has wird es in nullBits gecheckt. Sowie ich es richtig  verstanden habe ´, wenn ein Event nicht bei nullBits ist dann sollte es bei values zu finden sein

Reply:
@Njimefo @awildturtok das ist schon lustig. Da der BooloeanStore einen Tri-state speichern kann, hier aber nur einen bi-state gebraucht wird, gibt es die unschärfe

Reply:
Ich wollte eigentlich auch keinen bi-state, das ist aus einer Diskussion mit dir entstanden, leider weiß ich den Hintergrund nicht mehr. Die alte implementierung war glaube ich fixiert aus has=true und default war halt false